tests/bgp/bgp_helpers.py

```diff
@@ -30,22 +29,16 @@ def apply_bgp_config(duthost, template_name):
         duthost: DUT host object
         template_name: pathname of the bgp config on the DUT
     """
-    duthost.shell('docker cp {} {}'.format(template_name, DEFAULT_BGP_CONFIG))
-    restart_bgp(duthost)
-
-
-def restart_bgp(duthost, asic_index=DEFAULT_ASIC_ID):
-    """
-    Restart bgp services on the DUT
-
-    Args:
-        duthost: DUT host object
-    """
-    duthost.asic_instance(asic_index).reset_service("bgp")
-    duthost.asic_instance(asic_index).restart_service("bgp")
-    docker_name = duthost.asic_instance(asic_index).get_docker_name("bgp")
-    pytest_assert(wait_until(100, 10, 0, duthost.is_service_fully_started, docker_name), "BGP not started.")
-
+    duthost.docker_copy_to_all_asics('bgp', template_name, DEFAULT_BGP_CONFIG)
+    duthost.restart_service("bgp")
+    # Check if all BGPs are up
+    if duthost.is_multi_asic:
+        for asic_index in range(duthost.facts["num_asic"]):
+            docker_name = duthost.asic_instance(asic_index).get_docker_name("bgp")
+            pytest_assert(wait_until(100, 10, 0, duthost.is_service_fully_started, docker_name), "BGP not started.")
+    else:
+        pytest_assert(wait_until(100, 10, 0, duthost.is_service_fully_started, "bgp"), "BGP not started.") 
+  
```

Comment:
Can we add this block as function in multi_asic.py ?

Reply:
in multi_asic.py it is calling itself or the function in sonic_asic.py, to achieve that will need to add assert in the 2 restart_service() in the 2 files, but other functions do not have these checks, wonder if it is needed here?

    def restart_service(self, service):
        if service in self._DEFAULT_SERVICES:
            return self.sonichost.restart_service(service, service)
        for asic in self.asics:
            asic.restart_service(service)